<?xml version="1.0"?>
<robot name="URA">

    <link name="base_link">
        <visual name="base_viz">
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <cylinder radius="0.075" length="0.05" />
            </geometry>
            <material name="White">
                <color rgba="1 1 1 1"/>
            </material>
        </visual>
    </link>

    <link name="base_plate">
        <visual name="base_plate_viz">
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <cylinder radius="0.075" length="0.01" />
            </geometry>
            <material name="Blue">
                <color rgba="0 0 1 1"/>
            </material>
        </visual>
    </link>

    <link name="bottom_arm">
        <visual name="bottom_arm_viz">
            <origin xyz="0.1 0 0" rpy="0 0 0" />
            <geometry>
                <box size="0.20 0.002 0.045" />
            </geometry>
            <material name="Yellow">
                <color rgba="1 1 0 1"/>
            </material>
        </visual>
    </link>

    <link name="top_arm">
        <visual name="top_arm_viz">
            <origin xyz="0.1 0 0" rpy="0 0 0" />
            <geometry>
                <box size="0.20 0.002 0.045" />
            </geometry>
            <material name="Red">
                <color rgba="1 0 0 1"/>
            </material>
        </visual>
    </link>

    <link name="tool">
        <visual name="top_arm_viz">
            <origin xyz="0 -0.1 0" rpy="0 0 0" />
            <geometry>
                <box size="0.1 0.05 0.05" />
            </geometry>
            <material name="Lime">
                <color rgba="0 1 0 1"/>
            </material>
        </visual>
    </link>

    <joint name="base_rotation" type="revolute">
        <parent link="base_link"/>
        <child link="base_plate"/>
        <origin xyz="0 0 0.03" rpy="0 0 0" />
        <axis xyz="0 0 1"/>
        <limit lower="-3.14" upper="3.14" effort="10" velocity="10"/>
        <dynamics damping="50.0" friction="1.0"/>
    </joint>

    <joint name="bottom_rotation" type="revolute">
        <parent link="base_plate"/>
        <child link="bottom_arm"/>
        <origin xyz="0.03 0 0.045" rpy="0 0 0" />
        <axis xyz="0 1 0"/>
        <limit lower="0" upper="-1.57" effort="10" velocity="10"/>
        <dynamics damping="50.0" friction="1.0"/>
    </joint>

    <joint name="top_rotation" type="revolute">
        <parent link="bottom_arm"/>
        <child link="top_arm"/>
        <origin xyz="0.2 0 0" rpy="0 0 0" />
        <axis xyz="0 1 0"/>
        <limit lower="-1.57" upper="1.57" effort="10" velocity="10"/>
        <dynamics damping="50.0" friction="1.0"/>
    </joint>

    <joint name="tool_rotation" type="revolute">
        <parent link="top_arm"/>
        <child link="tool"/>
        <origin xyz="0.125 0 0" rpy="0 0 1.570796" />
        <axis xyz="0 1 0"/>
        <limit lower="-3.14" upper="3.14" effort="10" velocity="10"/>
        <dynamics damping="0.0" friction="0.0"/>
    </joint>

</robot>
